crypto/core_algorithm.c

```diff
@@ -65,6 +65,7 @@ static int algorithm_do_this(OSSL_PROVIDER *provider, void *cbdata)
                 data->fn(provider, thismap, no_store, data->data);
             }
         }
+        ossl_provider_query_free(provider, map);
```

Comment:
Are we inside a lock at this point? We must never have more than one query call before the unquery call.

Reply:
If `ossl_provider_query_operation` would be seen as an allocation, then I assume that the pointer would be unique where the provider does this, and locking shouldn't be an issue.

However, that makes for a *lot* of memory handling, as this is called every time we fetch anything.

Reply:
... that's another factor that has me thing the malloc()/free() view is not so great.

Reply:
The provider side is free to reference count if it returns the same array multiple times...